rcl/include/rcl/timer.h

```diff
@@ -137,7 +137,7 @@ RCL_WARN_UNUSED
 rcl_ret_t
 rcl_timer_init(
   rcl_timer_t * timer,
-  uint64_t period,
+  int64_t period,
```

Comment:
Should the passed period be checked to ensure it is >= 0?

Reply:
I guess so, I'm not sure what a negative period would indicate. I suppose it could be lumped in with zero, but my instinct would be to only allow zero as a special case and raise an error when the period is zero.

Reply:
> my instinct would be to only allow zero as a special case and raise an error when the period is zero.

Thanks, done in #295.

Reply:
@wjwwood,

A negative period and frequency are well defined derived units: 
https://en.wikipedia.org/wiki/Negative_frequency

The problem with software timers is that we don't know in what direction the elctrons will flow ;)